### DISCUSSION TITLE
geomap: Support multiple routes

### DISCUSSION THREAD

**Discussion by abalmos** [grafana/grafana#55422] [Dashboards, Panels & Visualizations]
We have started replacing ThingsBoard with Grafana now that the geomap route layer is available (last blocking feature for us). It is common to visualize multiple devices on the same geomap (machine's working together on a farm). The number and names of the devices are dynamic. As far as I can tell, this is not possible in the current implementation.

Our closest attempt so far is using a Postgres data source with three columns: `device name`, `lat`, and `lon`. When we return this as a table, the plot functions. However, all devices are the same fixed color and all device routes are connected with a line (from route to route).

I suppose I would have expected that each device be assigned its own color and that each route would be independently drawn on the map—at least this is our ThingsBoard mental model. I guess with Grafana some sort of group by on the device name would be needed? Looking at the current route layer implementation, I am unclear how I could properly contribute this functionality. Does this fit within the goals/scope of https://github.com/grafana/grafana/issues/44240? Would someone be willing to mentor us in implementing it?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
